ledger/src/blockstore.rs

```diff
@@ -1283,6 +1285,18 @@ impl Blockstore {
                     return false;
                 }
             }
+
+            // Check that the chaining between our current shred, the previous fec_set
+            // and the next fec_set
+            if !self.check_chained_merkle_root_consistency(
```

Comment:
This check (and similar other instances below), should only be invoked if a **new** erasure meta is inserted into blockstore. Otherwise we are redundantly repeating this check for every shred in the erasure batch.

Maybe we should also add the `chained_merkle_root` to `MerkleRootMeta` to check if the shreds within the same batch have the same `merkle_root` AND the same `chained_merkle_root`.

Reply:
that's a good point, i'll rework this to only check on new erasure metas. 

> Maybe we should also add the chained_merkle_root to MerkleRootMeta to check if the shreds within the same batch have the same merkle_root AND the same chained_merkle_root.

The `chained_merkle_root` is also part of the `merkle_root` right? so if the batch has the same `merkle_root` they also have the same `chained_merkle_root`?

Reply:
You are right. Just comparing `merkle_root`s is sufficient.